### PR TITLE
Add linkedlist_deque for typescript and javascript

### PR DESCRIPTION
**Pull request by zhuoqinyue**
If this PR is related to coding or code translation, please read the [contribution guideline](https://github.com/krahets/hello-algo/issues/15#issue-1464069591), fill out the checklist, and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: codes/javascript/chapter_stack_and_queue/linkedlist_deque.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
+/**
+ * File: linkedlist_deque.js
+ * Created Time: 2023-02-04
+ * Author: Zhuo Qinyue (1403450829@qq.com)
+ */
+
+class DequeNode {
```
Comment (krahets) [inline codes/javascript/chapter_stack_and_queue/linkedlist_deque.js:7]: /* 双向链表结点 */
ListNode

File: codes/typescript/chapter_stack_and_queue/linkedlist_deque.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
+/**
+ * File: linkedlist_deque.ts
+ * Created Time: 2023-02-04
+ * Author: Zhuo Qinyue (1403450829@qq.com)
+ */
+
+class DequeNode {
```
Comment (krahets) [inline codes/typescript/chapter_stack_and_queue/linkedlist_deque.ts:7]: /* 双向链表结点 */
ListNode

File: docs/chapter_stack_and_queue/deque.md [hunk 3]
```diff
@@ -149,13 +149,77 @@ comments: true
...
+    console.log("双向队列长度 size = " + size);
+
+    /* 判断双向队列是否为空 */
+    const isEmpty = size === 0;
+    console.log("双向队列是否为空 = " + isEmpty);
     ```
 
 === "TypeScript"
 
     ```typescript title="deque.ts"
```
Comment (krahets) [inline docs/chapter_stack_and_queue/deque.md:189]: Please add the `linkedlist_deque.js` and `linkedlist_deque.ts` to the docs as well.

**Review by krahets — COMMENTED** [review]
Thanks! Please address the comments.

File: codes/javascript/chapter_stack_and_queue/linkedlist_deque.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    constructor(val) {
+        this.val = val;
+        this.next = null;
+        this.prev = null;
+    }
+}
+
+/* 基于双向链表实现的双向队列 */
+class LinkedListDeque {
+    head;   // 头结点 head
```
Comment (krahets) [inline codes/javascript/chapter_stack_and_queue/linkedlist_deque.js:21]: front

File: codes/javascript/chapter_stack_and_queue/linkedlist_deque.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    tail;   // 尾结点 tail
```
Comment (krahets) [inline codes/javascript/chapter_stack_and_queue/linkedlist_deque.js:22]: rear

File: codes/javascript/chapter_stack_and_queue/linkedlist_deque.js [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+    len;    // 双向队列的长度
```
Comment (krahets) [inline codes/javascript/chapter_stack_and_queue/linkedlist_deque.js:23]: size

File: codes/javascript/chapter_stack_and_queue/linkedlist_deque.js [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+            // 将 node 添加至链表尾部
+            this.tail.next = node;
+            node.prev = this.tail;
+            this.tail = node; // 更新尾结点
+        }
+        this.len++;
+    }
+
+    /* 队首入队操作 */
+    pushFront(val) {
```
Comment (krahets) [inline codes/javascript/chapter_stack_and_queue/linkedlist_deque.js:48]: pushFirst

Please keep the code consistency with the reference code.

File: docs/chapter_stack_and_queue/deque.md (hunk 3 above)
Comment (zhuoqinyue) [reply docs/chapter_stack_and_queue/deque.md:189]: OK.
And there is a phenomenon.
I recommend this book to people around me, but some example codes are missing in the book, such as C language.
Any solution to this phenomenon? @krahets

File: codes/javascript/chapter_stack_and_queue/linkedlist_deque.js (hunk 6 above)
Comment (zhuoqinyue) [reply codes/javascript/chapter_stack_and_queue/linkedlist_deque.js:23]: Properties and methods in the js class cannot have the same name, so I use `len`.
If I name the property `size`, then I will name the method `getSize()`

File: codes/javascript/chapter_stack_and_queue/linkedlist_deque.js (hunk 7 above)
Comment (zhuoqinyue) [reply codes/javascript/chapter_stack_and_queue/linkedlist_deque.js:48]: Got it

**Comment by zhuoqinyue** [comment]
It's OK. @krahets

File: docs/chapter_stack_and_queue/deque.md (hunk 3 above)
Comment (krahets) [reply docs/chapter_stack_and_queue/deque.md:189]: C codes will be released after testing. Currently, we can only read C++ instead.

**Review by krahets — APPROVED** [review]
Thanks @zhuoqinyue !

**Pull request merged**